### PR TITLE
Laravel5 integration fixed

### PR DESCRIPTION
**Pull request by cmosguy**
Hey @ShawnMcCool and @mitchellvanw ,

I just completed my fix and contribution to getting the backup-manager working in Laravel 5.  I am working on a plugin for October CMS (which is all Laravel 5 based).  So, now this package should be good to go to work for the Laravel 5.  

You can see that I also updated the readme to support Laravel 4 and 5.  In Laravel 5 Taylor changed how packages are registered and published within the boot method. In addition, the config files are published with a new command in L5 called "vendor:publish".

I hope you accept this and merge and release into your master.  Let me know how this goes!

Thanks,
Adam

### CODE REVIEW

File: src/Integrations/Laravel/BackupManagerServiceProviderLaravel5.php [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+
+    /**
+     * Register the service provider.
+     *
+     * @return void
+     */
+    public function register()
+    {
+
+        $this->mergeConfigFrom(config_path('backup-manager/storage.php'), 'backup-manager::storage');
```
Comment (GrahamCampbell) [inline src/Integrations/Laravel/BackupManagerServiceProviderLaravel5.php]: This should actually be done in boot.

File: src/Integrations/Laravel/BackupManagerServiceProviderLaravel5.php [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+        $this->mergeConfigFrom(config_path('backup-manager/database.php'), 'backup-manager::database');
```
Comment (GrahamCampbell) [inline src/Integrations/Laravel/BackupManagerServiceProviderLaravel5.php]: This should actually be done in boot.

Comment (GrahamCampbell) [inline src/Integrations/Laravel/BackupManagerServiceProviderLaravel5.php]: Also, your 2nd param here is incorrect.

File: src/Integrations/Laravel/BackupManagerServiceProviderLaravel5.php (hunk 1 above)
Comment (cmosguy) [inline src/Integrations/Laravel/BackupManagerServiceProviderLaravel5.php]: Hi @GrahamCampbell ,

I am not sure I understand, can you tell me which line you mean specifically?  In Taylor's [docs](http://laravel.com/docs/5.0/packages#configuration) that "To publish a configuration file, just use the publishes method from the boot method of your service provider:"  and "To merge the configurations, use the mergeConfigFrom method within your service provider's register method".  Can you please advise further what the code should be?  I tested it and it seems to be working.

File: src/Integrations/Laravel/BackupManagerServiceProviderLaravel5.php (hunk 2 above)
Comment (cmosguy) [inline src/Integrations/Laravel/BackupManagerServiceProviderLaravel5.php]: @GrahamCampbell what do you recommend that the second param be?  I was trying not to muck with @ShawnMcCool code for the line:

```php
$provider = new Filesystems\FilesystemProvider(new Config($app['config']['backup-manager::storage']));
```

Again, can you please advise what you recommend here?  If you test this out you can see that it works during database backup and restore.  I tested only for AWS S3 bucket so far.

**Comment by cmosguy** [comment]
@GrahamCampbell , @ShawnMcCool , @mitchellvanw , i had to do some more updates.  It seems that Laravel 5 config system is not very friendly with subdirectories. So, for Laravel 5 specifically, I have made the requirement for the user to just put the stuff in `config/backup-manager.php`.  This is how Laravel 5 configs work and are more explicit.  It is also more compatible with the other systems like October CMS.

Please see if you can merge the PR soon in the master, it would be great to get things rolling here.  Let me know your thoughts.

Thanks!
Adam

**Comment by ShawnMcCool** [comment]
Hmm.. We need to move the Laravel driver out of this package. I like the idea of having a laravel 4 repo and a laravel 5 repo that both depend on this repository.

**Comment by cmosguy** [comment]
@ShawnMcCool ok, what do you recommend, do you want to create another repo for this and take ownership of it?

**Comment by ShawnMcCool** [comment]
I've moved this package to a new organization. I've created framework-version-specific repositories for drivers.

https://github.com/backup-manager

**Comment by ShawnMcCool** [comment]
@cmosguy ok! 1.0 of the backup manager is done. (which means i removed laravel support). Additionally, laravel support was added here https://github.com/backup-manager/laravel-4

I would like to have https://github.com/backup-manager/laravel-5 set up specifically for laravel 5. Could you help with this?

File: README.md [hunk 3]
```diff
@@ -178,8 +179,27 @@ The required bootstrapping can [be found in the example here](https://github.com
...
+
+```php
+'BigName\BackupManager\Integrations\Laravel\BackupManagerServiceProviderLaravel5',
+```
+
+Then, publish and modify the configuration file to suit your needs.
+
+`php artisan vendor:publish`
+
+Finally, you can edit the `config/backup-manager.php` with the proper storage settings.  An example is shown above in the [storage.php](#storage-php) example contents, but remember makes ure the file is called `backup-manager.php`.
```
Comment (frankdejonge) [inline README.md]: make use => make sure

**Comment by ShawnMcCool** [comment]
Ok, actually, I just created a single Laravel driver repository 

http://github.com/backup-manager/laravel

We can add a new service provider for laravel 5 specifically.

**Comment by ShawnMcCool** [comment]
@cmosguy Ok, so I created this file https://github.com/backup-manager/laravel/blob/master/src/Laravel5BackupManagerServiceProvider.php

It's a copy of the Laravel 4 provider, could you perhaps push your changes against it?

**Comment by cmosguy** [comment]
@ShawnMcCool just tried to do a fresh install of Laravel 5.1, with the back-up manager:

```
  Problem 1
    - Installation request for backup-manager/laravel dev-master -> satisfiable by backup-manager/laravel[dev-master].
    - Conclusion: remove laravel/framework v5.1.2
    - Conclusion: don't install laravel/framework v5.1.2
    - backup-manager/laravel dev-master requires illuminate/console 4.x -> satisfiable by illuminate/console[4.0.x-dev,
4.1.x-dev, 4.2.x-dev, v4.0.0, v4.0.0-BETA2, v4.0.0-BETA3, v4.0.0-BETA4, v4.0.1, v4.0.10, v4.0.2, v4.0.3, v4.0.4, v4.0.5,
 v4.0.6, v4.0.7, v4.0.8, v4.0.9, v4.1.0, v4.1.1, v4.1.10, v4.1.11, v4.1.12, v4.1.13, v4.1.14, v4.1.15, v4.1.16, v4.1.17,
 v4.1.18, v4.1.19, v4.1.2, v4.1.20, v4.1.21, v4.1.22, v4.1.23, v4.1.24, v4.1.25, v4.1.26, v4.1.27, v4.1.28, v4.1.29, v4.
1.3, v4.1.30, v4.1.4, v4.1.5, v4.1.6, v4.1.7, v4.1.8, v4.1.9, v4.2.0-BETA1, v4.2.1, v4.2.12, v4.2.16, v4.2.17, v4.2.2, v
4.2.3, v4.2.4, v4.2.5, v4.2.6, v4.2.7, v4.2.8, v4.2.9].
    - don't install illuminate/console 4.0.x-dev|don't install laravel/framework v5.1.2
    - don't install illuminate/console 4.1.x-dev|don't install laravel/framework v5.1.2
    - don't install illuminate/console 4.2.x-dev|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.0.0|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.0.0-BETA2|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.0.0-BETA3|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.0.0-BETA4|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.0.1|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.0.10|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.0.2|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.0.3|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.0.4|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.0.5|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.0.6|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.0.7|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.0.8|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.0.9|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.1.0|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.1.1|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.1.10|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.1.11|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.1.12|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.1.13|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.1.14|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.1.15|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.1.16|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.1.17|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.1.18|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.1.19|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.1.2|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.1.20|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.1.21|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.1.22|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.1.23|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.1.24|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.1.25|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.1.26|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.1.27|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.1.28|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.1.29|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.1.3|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.1.30|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.1.4|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.1.5|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.1.6|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.1.7|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.1.8|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.1.9|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.2.0-BETA1|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.2.1|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.2.12|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.2.16|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.2.17|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.2.2|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.2.3|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.2.4|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.2.5|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.2.6|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.2.7|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.2.8|don't install laravel/framework v5.1.2
    - don't install illuminate/console v4.2.9|don't install laravel/framework v5.1.2
    - Installation request for laravel/framework == 5.1.2.0 -> satisfiable by laravel/framework[v5.1.2].
```

It looks like there are some dependencies that need to be resolved first.

File: composer.json [hunk 4]
```diff
@@ -22,9 +22,10 @@
   "require-dev": {
     "mockery/mockery": "dev-master",
     "satooshi/php-coveralls": "~0.6",
-    "illuminate/support": "4.x",
-    "illuminate/container": "4.x",
-    "illuminate/console": "4.x",
+    "illuminate/container": "5.x",
```
Comment (GrahamCampbell) [inline composer.json]: :-1:

File: src/Integrations/Laravel/BackupManagerServiceProviderLaravel5.php [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+    protected $defer = true;
+
+    /**
+     * Bootstrap the application events.
+     *
+     * @return void
+     */
+    public function boot()
+    {
+        //$this->package('heybigname/backup-manager', 'backup-manager', __DIR__.'/../../..');
```
Comment (GrahamCampbell) [inline src/Integrations/Laravel/BackupManagerServiceProviderLaravel5.php]: Never comment out code, just delete it.

File: config/storage.php [hunk 6]
```diff
@@ -3,13 +3,13 @@
...
-        'root' => '/',
+        'root' => base_path('/storage/app/backups'),
     ],
     's3' => [
         'type' => 'AwsS3',
         'key'    => '',
-        'secret' => '',
-        'region' => Aws\Common\Enum\Region::US_EAST_1,
+        'secret' => '//',
+        'region' => 'us-east-1',
```
Comment (ShawnMcCool) [inline config/storage.php]: Constants are a more error-resistant alternative to strings.

Comment (GrahamCampbell) [inline config/storage.php]: enums are removed from the aws sdk v3

Comment (GrahamCampbell) [inline config/storage.php]: nb, laravel 5.1 needs the aws sdk v3

Comment (ShawnMcCool) [inline config/storage.php]: I swear.. What is the PHP world's obsession with strings?

**Comment by ShawnMcCool** [comment]
@cmosguy I have taken the code here and push it into the Laravel driver. Please post all changes there.

**Pull request closed without merging**